src/Alpacc/RegularExpression.hs

```diff
@@ -0,0 +1,67 @@
+module Alpacc.RegularExpression (regExFromText, pRegEx) where
+
+import Data.Composition
+import Data.Text (Text)
+import Data.Void
+import Text.Megaparsec
+import Text.Megaparsec.Char (char, space1, string)
+import Text.Megaparsec.Char.Lexer qualified as Lexer
+
+type Parser = Parsec Void Text
+
+data RegEx
+  = Epsilon
+  | Literal Char
+  | Star RegEx
+  | Alter RegEx RegEx
+  | Concat RegEx RegEx
+  deriving (Eq, Show)
+
+space :: Parser ()
+space = Lexer.space space1 (Lexer.skipLineComment "#") empty
+
+lexeme :: Parser a -> Parser a
+lexeme = Lexer.lexeme space
+
+pLiteral :: Parser RegEx
+pLiteral = Literal <$> lexeme (satisfy (`elem` ['a' .. 'z']))
+
+chainl1 :: Parser a -> Parser (a -> a -> a) -> Parser a
+chainl1 p op = p >>= rest
+  where
+    rest x =
+      do
+        f <- op
+        y <- p
+        rest (f x y)
+        <|> return x
+
+type BinaryOp = RegEx -> RegEx -> RegEx
+
+binaryOps :: [Text] -> [BinaryOp] -> Parser BinaryOp
+binaryOps = (lexeme . foldl1 (<|>)) .: zipWith toBinaryOp
+  where
+    toBinaryOp x f = string x >> return f
+
+pConcat :: Parser RegEx
+pConcat = pTerm `chainl1` binaryOps [""] [Concat]
```

Comment:
Don't use empty strings as operators.  I think `pConcat` should just be `many pTerm`.

Reply:
I think I am using many correctly now.